### PR TITLE
NBSNEBIUS-356: retry io when io_submit return EAGAIN

### PR DESCRIPTION
**Pull request by budevg**
We discovered cases where aio_submit return with EAGAIN
This can happen when aio subsystem is temporarily out of requests for example
here https://elixir.bootlin.com/linux/latest/source/fs/aio.c#L2054

In this commit we start to handle EAGAIN as retryable error to avoid marking
devices as broken

### CODE REVIEW

File: cloud/blockstore/libs/storage/disk_agent/storage_with_stats.cpp [hunk 1]
```diff
@@ -135,6 +158,12 @@ struct TStorageWithIoStats final
     {
         Stats->OnError();
     }
+
+    void HandleIoError(ResponseCode rc) {
```
Comment (qkrorlqr) [inline cloud/blockstore/libs/storage/disk_agent/storage_with_stats.cpp:162]: { не там

File: cloud/blockstore/libs/storage/disk_agent/storage_with_stats.cpp [hunk 2]
```diff
@@ -13,10 +13,27 @@ namespace {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+enum class ResponseCode
```
Comment (qkrorlqr) [inline cloud/blockstore/libs/storage/disk_agent/storage_with_stats.cpp:16]: EResponseCode

**Comment by qkrorlqr** [comment]
какой-то тест нужен

File: cloud/blockstore/libs/storage/disk_agent/storage_with_stats.cpp [hunk 3]
```diff
@@ -13,10 +13,27 @@ namespace {
...
+        return ResponseCode::Error;
+    }
+
+    auto& error = future.GetValue().GetError();
+    if (HasError(error)) {
+        return error.GetCode() == E_REJECTED ? ResponseCode::Retry
+                                             : ResponseCode::Error;
+    }
+
+    return ResponseCode::Success;
```
Comment (sharpeye) [inline cloud/blockstore/libs/storage/disk_agent/storage_with_stats.cpp:36]: Можно заюзать фичи TError:
```cpp
template <typename T>
ui32 GetResponseCode(const TFuture<T>& future)
{
    if (future.HasException()) {
        return E_FAIL;
    }

    return future.GetValue().GetError().GetCode();
}


// 

if (HandleIoError(GetResponseCode(future))) {
    return;
}

// 

bool HandleIoError(ui32 code)
{
    if (FAILED(code) &&
        GetErrorKind(MakeError(code)) != EErrorKind::ErrorRetriable)
    {
        Stats->OnError();

        return true;
    }

    return false;
}
```

**Review by drbasic — APPROVED** [review]

**Review by sharpeye — APPROVED** [review]

**Review by qkrorlqr — APPROVED** [review]

**Pull request merged**